### ISSUE TITLE
SpaceAfterMethodName cop throws error when using grouped expression as method param

### ISSUE THREAD

**Issue by kbackowski** [bbatsov/rubocop#1739]
Having following code :

```
class Test
  def self.test(value)
    value
  end
end

Test.test ('0' || '1').to_i
```

Rubocop will throw error :

```
Inspecting 1 file
W

Offenses:

test.rb:7:10: W: (...) interpreted as grouped expression.
Test.test ('0' || '1').to_i
         ^

1 file inspected, 1 offense detected
```

In this case SpaceAfterMethodName should not throw any error since parenthesis here are required for calling to_i on the evaluated value.

```
rubocop -V
0.29.1 (using Parser 2.2.0.3, running on ruby 2.2.1 x86_64-linux)
```

**Comment by jonas054** [comment]
Run `rubocop` with the `-D` / `--display-cop-names` flag and you'll see that it's not the `SpaceAfterMethodName` that reports the offense - it's `Lint/ParenthesesAsGroupedExpression`. The solution is to add parentheses for the method call:
```ruby
Test.test(('0' || '1').to_i)
```
`ruby -w` gives the same warning, BTW.

**Comment by kbackowski** [comment]
Indeed the warning is being reported by the Lint/ParenthesesAsGroupedExpression. However newest ruby doesn't throw any warnings for this syntax :

```
$ ruby -cw test.rb
Syntax OK
$ ruby --version
ruby 2.2.1p85 (2015-02-26 revision 49769) [x86_64-linux]
``` 
So i'm not sure if the additional parenthesis are really required in here ?

**Comment by jonas054** [comment]
The warning means "Maybe you meant `(Test.test('0' || '1')).to_i`" so I think it's a good warning, and I don't know why it's been removed in Ruby 2.2.

**Comment by kbackowski** [comment]
Ok, there's a rule for this in the ruby style guide (https://github.com/bbatsov/ruby-style-guide#parens-as-args) so it should be a good practice to use additonal parenthesis in this case.

Thanks for clarification
